odd_collector/adapters/databricks/mappers/column.py

```diff
@@ -11,18 +17,134 @@ def split_by_braces(value: str) -> str:
+            )
+            for field_name, field_type in column_type.fields.items():
+                _build_ds_field_from_type(field_name, field_type, oddrn)
+        elif isinstance(column_type, Map):
+            generated_dataset_fields.append(
+                DataSetField(
+                    oddrn=oddrn,
+                    name=column_name,
+                    metadata=[
+                        extract_metadata(
+                            "databricks", column, DefinitionType.DATASET_FIELD
+                        )
+                    ],
+                    type=DataSetFieldType(
+                        type=Type.TYPE_MAP,
+                        logical_type=get_logical_type(column_type),
+                        is_nullable=False,
+                    ),
+                    owner=None,
+                    parent_field_oddrn=parent_oddrn,
+                )
+            )
+            _build_ds_field_from_type("Key", column_type.key_type, oddrn)
+            _build_ds_field_from_type("Value", column_type.value_type, oddrn)
+
+        else:
+            odd_type = get_databricks_type(column_type)
+            logical_type = get_logical_type(column_type)
+            logger.debug(
+                f"Column {column_name} has ODD type {odd_type} and logical type {logical_type}"
+            )
+            generated_dataset_fields.append(
+                DataSetField(
+                    oddrn=oddrn,
+                    name=column_name,
+                    metadata=[
+                        extract_metadata(
+                            "databricks", column, DefinitionType.DATASET_FIELD
+                        )
+                    ],
+                    type=DataSetFieldType(
+                        type=odd_type,
+                        logical_type=logical_type,
+                        is_nullable=False,
+                    ),
+                    owner=None,
+                    parent_field_oddrn=parent_oddrn,
+                )
+            )
+
+    _build_ds_field_from_type(column.name, column_type)
+    return generated_dataset_fields
+
+
+def get_logical_type(type_field: Union[ParseType, str]) -> str:
+    if isinstance(type_field, BasicType):
+        return type_field.type_name
+    elif isinstance(type_field, ArrayType):
+        return f"Array({get_logical_type(type_field.type)})"
+    elif isinstance(type_field, Map):
+        return f"Map({get_logical_type(type_field.key_type)}, {get_logical_type(type_field.value_type)})"
+    elif isinstance(type_field, str):
+        return type_field
+    elif isinstance(type_field, Struct):
+        return (
+            "Struct("
+            + ", ".join(
+                f"{name}: {get_logical_type(type)}"
+                for name, type in type_field.fields.items()
+            )
+            + ")"
+        )
+    else:
+        raise UndefinedType(f"Undefiend type {type_field}")
+
+
+def get_databricks_type(type_field: Union[ParseType, str]) -> Type:
```

Comment:
It returns ODD Type, should it be better to rename to something like `get_odd_type`?